threadit-api/Controllers/v1/CommentsController.cs

```diff
@@ -0,0 +1,78 @@
+using System.Net.Cache;
+using ThreaditAPI.Extensions;
+using ThreaditAPI.Middleware;
+using Microsoft.AspNetCore.Mvc;
+using ThreaditAPI.Models;
+using ThreaditAPI.Models.Requests;
+using ThreaditAPI.Services;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace ThreaditAPI.Controllers.v1 {
+    [ApiController]
+    [Route("v1/comments")]
+    public class CommentsController : ControllerBase {
+        
+        [HttpGet("{threadId}")]
+        public async Task<IActionResult> GetBaseComments([FromRoute] string threadId, [FromServices] CommentService commentService) {    
+            CommentFull[] comments = await commentService.GetBaseComments(threadId);
+            return Ok(comments);
+        }
+
+        [HttpGet("{threadId}/expand/{parentCommentId}")]
+        public async Task<IActionResult> ExpandComments([FromRoute] string threadId, [FromRoute] string parentCommentId, [FromServices] CommentService commentService) {            
+            CommentFull[] comments = await commentService.ExpandComments(threadId, parentCommentId);
+            return Ok(comments);
+        }
+
+        [HttpGet("{threadId}/older/{siblingCommentId}")]
+        public async Task<IActionResult> OlderComments([FromRoute] string threadId, [FromRoute] string siblingCommentId, [FromServices] CommentService commentService) {   
+            CommentFull[] comments = await commentService.OlderComments(threadId, siblingCommentId);
+            return Ok(comments);
+        }
+
+        [HttpGet("{threadId}/newer/{siblingCommentId}")]
+        public async Task<IActionResult> NewerComments([FromRoute] string threadId, [FromRoute] string siblingCommentId, [FromServices] CommentService commentService) {            
+            CommentFull[] comments = await commentService.NewerComments(threadId, siblingCommentId);
+            return Ok(comments);
+        }
+
+        [HttpPost("{threadId}/{parentCommentId?}")]
```

Comment:
can we not wrap this and the first function into the same since parentCommentId is nullable?